calico_node/filesystem/startup.py

```diff
@@ -259,9 +260,10 @@ def main():
     # Write a startup environment file containing the IP address that may have
     # just been detected.
     # This is required because the confd templates expect to be able to fill in
-    #  the IP by fetching it from the environment.
+    # some templates by fetching them from the environment.
     with open('startup.env', 'w') as f:
-        f.write("IP=" + ip)
+        f.write("export IP=%s\n" % ip)
```

Comment:
Various scripts source this fie.  However, since the values are not exported they don't propagate to child processes, meaning things like `confd` which rely on these values being set in the environment don't see them.

@matthewdupre WDYT?